### ISSUE TITLE
Minio chart 7.1.2 on Kubernetes 1.17.5

### ISSUE THREAD

**Issue by clemh78** [minio/charts#37]
Hello,

I have an error when i want to upgrade Minio on Kubernetes 1.17.5.

This is error i have when i upgrade with helm command.

```
helm3 upgrade minio minio/minio -f values.yaml -n minio --version 7.1.2

Error: UPGRADE FAILED: current release manifest contains removed kubernetes api(s) for this kubernetes version and it is therefore unable to build the kubernetes objects for performing the diff. error from kubernetes: unable to recognize "": no matches for kind "StatefulSet" in version "apps/v1beta2"
```

I dont understand this error because you have take care of the new version of Kubernetes with this line : https://github.com/minio/charts/blob/5bb5fe5f2c67e69d9b436f95511c3a0252cdb759/minio/templates/_helpers.tpl#L60

Have you an idea of the reason of this error ?

**Comment by Fabrice-44** [comment]
Hello,
I have a similar error with my helm upgrade, except that I'm on Kubernetes v1.16.9
```
helm upgrade --install minio -f my-values.yml --namespace minio --version 7.1.2

Error: UPGRADE FAILED: unable to recognize "": no matches for kind "StatefulSet" in version "apps/v1beta2"
```

I think there is an error in the helpers file about this test (and about kube version) :
```
{{- if semverCompare "<1.17-0" .Capabilities.KubeVersion.GitVersion -}}
{{- print "apps/v1beta2" -}}
```
Does the version should not be "1.16-0" here rather than 1.17 ? (so that people who are in version 1.16 & 1.17 have to use "apps/v1"

**Comment by WMP** [comment]
https://github.com/minio/charts/pull/43

**Comment by harshavardhana** [comment]
Fixed by #52
